src/composable/chat/useChatMessages.ts

```diff
@@ -258,6 +258,17 @@ export function useChatMessages(ctx: ChannelContext) {
 	 */
 	async function awaitMessage(id: string, timeout = 1e4): Promise<ChatMessage> {
 		return new Promise((resolve, reject) => {
+			// Check displayed and buffer if message has already been pushed.
+			const displayedMessage = data.displayed.find((msg) => {
+				return msg.id === id;
+			});
+			if (displayedMessage) resolve(displayedMessage);
```

Comment:
You should return here, otherwise the code below will still run

```suggestion
			if (displayedMessage) return resolve(displayedMessage);
```